kiva/agg/agg-24/font_freetype/agg_font_freetype.cpp

```diff
@@ -490,6 +490,7 @@ namespace agg24
 
 
 
+    static const unsigned _face_lookup_scratch_length = 1024;

```

Comment:
This is limiting the total length of the face name + face index in the lookup string?

Is there a possible failure mode where a user passes a _really_ long font name, even if it is bogus?

Reply:
Seemed like a reasonable value, but maybe [32k](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation) is better?

For performance reasons, I'd really like to avoid allocating this every time `find_face` is called (which is every time `load_font` is called, which can be a lot).

Reply:
Looks like TrueType font names can't be longer than a UInt16 (digging in here: https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6name.html).  So it depends - if this holds a path, then I guess max path length + a bit makes sense, otherwise 64k + a bit should be safe.

Reply:
This holds a path. I suppose the same "allocate a new buffer when the input grows" approach could be used here as what is used for the `m_signature` member variable in this same class.